### PR TITLE
docs: rfc: 0006-api

### PR DESCRIPTION
**Pull request by drmingdrmer**
##### docs: rfc: 0004-server-state

- - -
This change is [Reviewable](https://reviewable.io/reviews/suraft/suraft/5)

### CODE REVIEW

File: rfcs/0006-implementation-api.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Storage API
+
+**suRaft** requires a storage interface with the following functions:
+
+- **Write**:
+
+  ```rust
+  fn write(path: &str, buf: &[u8], put_if_absent: bool) -> Result<bool, io::Error>
```
Comment (tisonkun) [inline rfcs/0006-implementation-api.md:8]: This assumes all the payload must be able to hold in a buffer. Is it desired?

In OpenDAL, the read/write APIs are symmetrical:

* read(path) -> Result<Buffer>
* reader(path) -> Result<Reader>
* write(path, buf, [put_if_absent]) -> Result<()>
* writer(path, [put_if_absent]) -> Result<Writer>

File: rfcs/0006-implementation-api.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  Lists all paths in storage that start with the given `prefix`,  beginning after `start_after`.
+
+# Network API
+
+For communication between servers, **suRaft** defines the following network functions:
+
+- **Request Vote**:
+
+  ```rust
+  fn request_vote(&self, req: RequestVote) -> Result<RequestVoteReply, NetError>
```
Comment (tisonkun) [inline rfcs/0006-implementation-api.md:36]: `async fn`? Or the async part will be wrapped outside.

File: rfcs/0006-implementation-api.md (hunk 1 above)
Comment (drmingdrmer) [reply rfcs/0006-implementation-api.md:8]: Yes. the first version is a prototype with simpler API

File: rfcs/0006-implementation-api.md (hunk 2 above)
Comment (drmingdrmer) [reply rfcs/0006-implementation-api.md:36]: Right. It's an `async fn`...

**Pull request merged**